spec/statement.dd

```diff
@@ -491,6 +491,19 @@ foreach (int i, char c; a)
         order.
         )
 
+        $(P $(B Note:) The $(I ForeachTypeAttribute) is implicit, and when a type is not specified, it is inferred. In that case, $(D auto) is implied, and it is not necessary (and actually forbidden) to use it.
+        )
+
+--------------
+char[] a;
+...
+foreach(c; a) // ok the type of c is char
+    write(c,' ');
+
+foreach(auto c; a) // error
+    write(c,' ');
```

Comment:
I am pretty opinionated, but I would remove the "error" example - the other one makes it already clear.